File: python/2216.py
from typing import List

class Solution:
    def minDeletion(self, nums: List[int]) -> int:
        ans = 0
        i = 0
        while i < len(nums)-1:
            if nums[i] != nums[i+1]:
                i += 2
            else:
                i += 1
                ans += 1
        if (len(nums) - ans) % 2 == 1:
            ans += 1
        return ans

File: go/2216.go
package main

func minDeletion(nums []int) (ans int) {
	for i := 0; i < len(nums)-1; {
		if nums[i] != nums[i+1] {
			i += 2
		} else {
			i++
			ans++
		}
	}
	if (len(nums)-ans)%2 == 1 {
		ans++
	}
	return
}

// func main() {
// 	nums := []int{1, 1, 2, 2, 3, 3, 3}
// 	fmt.Printf("minDeletion(nums): %v\n", minDeletion(nums))
// }
